src/charm.py

```diff
@@ -18,3 +24,56 @@ def __init__(self, *args):
             args: Arguments passed to the CharmBase parent constructor.
         """
         super().__init__(*args)
+        self.framework.observe(self.on.config_changed, self._on_config_changed)
+        self.framework.observe(self.on.saml_relation_created, self._on_relation_created)
+
+    def _on_config_changed(self, _) -> None:
+        """Handle changes in configuration."""
+        self.unit.status = ops.MaintenanceStatus("Configuring charm")
+        try:
+            self._charm_state = CharmState.from_charm(charm=self)
```

Comment:
Should this be done in `__init__`?

Reply:
What's the reasoning behind that?

Reply:
Generally speaking, in Python classes, attributes should only be define in `__init__` so that it is clear what is and isn't available as an attribute on the class. There shouldn't be any new information in an event handler that is not available during `__init__` since it is called for every hook (except for deferred events which should be changed in the future)

Reply:
I've initialized them in the `__init__` method, Thx

Reply:
I can't see the change now, did you push it ?